libraries/chain/controller.cpp

```diff
@@ -3371,7 +3351,7 @@ struct controller_impl {
             auto existing = forkdb.get_block( id );
             EOS_ASSERT( !existing, fork_database_exception, "we already know about this block: ${id}", ("id", id) );
 
-            auto prev = forkdb.get_block_header( b->previous );
+            auto prev = forkdb.get_block( b->previous, true );
```

Comment:
Do you want to define an enum instead of `true` or `false`?

Reply:
done